File: packages/go-client-sdk/pkg/models/operations/queryseatsuserseat.go
package operations

import (
	"time"
)

type QuerySeatsUserSeatPathParams struct {
	SubscriptionID string `pathParam:"style=simple,explode=false,name=subscriptionId"`
	TenantID       string `pathParam:"style=simple,explode=false,name=tenantId"`
	UserID         string `pathParam:"style=simple,explode=false,name=userId"`
}

type QuerySeatsUserSeatRequest struct {
	PathParams QuerySeatsUserSeatPathParams
}

type QuerySeatsUserSeatDefaultApplicationJSONIssues struct {
	Message string `json:"message"`
}

type QuerySeatsUserSeatDefaultApplicationJSON struct {
	Code    string                                           `json:"code"`
	Issues  []QuerySeatsUserSeatDefaultApplicationJSONIssues `json:"issues,omitempty"`
	Message string                                           `json:"message"`
}

type QuerySeatsUserSeat200ApplicationJSONOccupant struct {
	Email    string `json:"email"`
	SeatID   string `json:"seat_id"`
	TenantID string `json:"tenant_id"`
	UserID   string `json:"user_id"`
	UserName string `json:"user_name"`
}

type QuerySeatsUserSeat200ApplicationJSONReservation struct {
	Email     string `json:"email"`
	InviteURL string `json:"invite_url"`
	SeatID    string `json:"seat_id"`
	TenantID  string `json:"tenant_id"`
	UserID    string `json:"user_id"`
}

type QuerySeatsUserSeat200ApplicationJSONSeatTypeEnum string

const (
	QuerySeatsUserSeat200ApplicationJSONSeatTypeEnumStandard QuerySeatsUserSeat200ApplicationJSONSeatTypeEnum = "standard"
	QuerySeatsUserSeat200ApplicationJSONSeatTypeEnumLimited  QuerySeatsUserSeat200ApplicationJSONSeatTypeEnum = "limited"
)

type QuerySeatsUserSeat200ApplicationJSONSeatingStrategyNameEnum string

const (
	QuerySeatsUserSeat200ApplicationJSONSeatingStrategyNameEnumMonthlyActiveUser    QuerySeatsUserSeat200ApplicationJSONSeatingStrategyNameEnum = "monthly_active_user"
	QuerySeatsUserSeat200ApplicationJSONSeatingStrategyNameEnumFirstComeFirstServed QuerySeatsUserSeat200ApplicationJSONSeatingStrategyNameEnum = "first_come_first_served"
)

type QuerySeatsUserSeat200ApplicationJSON struct {
	CreatedUtc          time.Time                                                   `json:"created_utc"`
	ExpiresUtc          time.Time                                                   `json:"expires_utc"`
	ID                  string                                                      `json:"id"`
	Occupant            QuerySeatsUserSeat200ApplicationJSONOccupant                `json:"occupant"`
	RedeemedUtc         time.Time                                                   `json:"redeemed_utc"`
	Reservation         QuerySeatsUserSeat200ApplicationJSONReservation             `json:"reservation"`
	SeatType            QuerySeatsUserSeat200ApplicationJSONSeatTypeEnum            `json:"seat_type"`
	SeatingStrategyName QuerySeatsUserSeat200ApplicationJSONSeatingStrategyNameEnum `json:"seating_strategy_name"`
	SubscriptionID      string                                                      `json:"subscription_id"`
}

type QuerySeatsUserSeatResponse struct {
	ContentType                                    string
	StatusCode                                     int64
	QuerySeatsUserSeat200ApplicationJSONObject     *QuerySeatsUserSeat200ApplicationJSON
	QuerySeatsUserSeatDefaultApplicationJSONObject *QuerySeatsUserSeatDefaultApplicationJSON
}

File: packages/ts-client-sdk/src/sdk/models/operations/queryseatsuserseat.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class QuerySeatsUserSeatPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscriptionId" })
  subscriptionId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=tenantId" })
  tenantId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=userId" })
  userId: string;
}

export class QuerySeatsUserSeatRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: QuerySeatsUserSeatPathParams;
}

export class QuerySeatsUserSeatDefaultApplicationJSONIssues extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export class QuerySeatsUserSeatDefaultApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=code" })
  code: string;

  @SpeakeasyMetadata({ data: "json, name=issues", elemType: QuerySeatsUserSeatDefaultApplicationJSONIssues })
  issues?: QuerySeatsUserSeatDefaultApplicationJSONIssues[];

  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export class QuerySeatsUserSeat200ApplicationJSONOccupant extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=email" })
  email: string;

  @SpeakeasyMetadata({ data: "json, name=seat_id" })
  seatId: string;

  @SpeakeasyMetadata({ data: "json, name=tenant_id" })
  tenantId: string;

  @SpeakeasyMetadata({ data: "json, name=user_id" })
  userId: string;

  @SpeakeasyMetadata({ data: "json, name=user_name" })
  userName: string;
}

export class QuerySeatsUserSeat200ApplicationJSONReservation extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=email" })
  email: string;

  @SpeakeasyMetadata({ data: "json, name=invite_url" })
  inviteUrl: string;

  @SpeakeasyMetadata({ data: "json, name=seat_id" })
  seatId: string;

  @SpeakeasyMetadata({ data: "json, name=tenant_id" })
  tenantId: string;

  @SpeakeasyMetadata({ data: "json, name=user_id" })
  userId: string;
}
export enum QuerySeatsUserSeat200ApplicationJSONSeatTypeEnum {
    Standard = "standard",
    Limited = "limited"
}
export enum QuerySeatsUserSeat200ApplicationJSONSeatingStrategyNameEnum {
    MonthlyActiveUser = "monthly_active_user",
    FirstComeFirstServed = "first_come_first_served"
}

export class QuerySeatsUserSeat200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=created_utc" })
  createdUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=expires_utc" })
  expiresUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=occupant" })
  occupant: QuerySeatsUserSeat200ApplicationJSONOccupant;

  @SpeakeasyMetadata({ data: "json, name=redeemed_utc" })
  redeemedUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=reservation" })
  reservation: QuerySeatsUserSeat200ApplicationJSONReservation;

  @SpeakeasyMetadata({ data: "json, name=seat_type" })
  seatType: QuerySeatsUserSeat200ApplicationJSONSeatTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=seating_strategy_name" })
  seatingStrategyName: QuerySeatsUserSeat200ApplicationJSONSeatingStrategyNameEnum;

  @SpeakeasyMetadata({ data: "json, name=subscription_id" })
  subscriptionId: string;
}

export class QuerySeatsUserSeatResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  querySeatsUserSeat200ApplicationJSONObject?: QuerySeatsUserSeat200ApplicationJSON;

  @SpeakeasyMetadata()
  querySeatsUserSeatDefaultApplicationJSONObject?: QuerySeatsUserSeatDefaultApplicationJSON;
}